ssntp/frame.go

```diff
@@ -291,3 +292,14 @@ func (f Frame) DumpTrace() (*payloads.FrameTrace, error) {
 
 	return &s, nil
 }
+
+// OriginUUID is the frame first sender and creator UUID.
+// When a SSNTP frame is forwarded by a server, the client
+// then only sees a new frame coming but it can not tell
+// who the frame creator and first sender is. This method
+// allows to fetch such information from a frame.
+func (f Frame) OriginUUID() (origin uuid.UUID) {
+	copy(origin[:], f.Origin[:16])
```

Comment:
Or just make f.Origin a uuid.UUID.  That way you'd be sure that the copy would always work and as a bonus you wouldn't need to change session.go.  It's probably more space efficient as well, but the semantics will be slightly different.  If the session.src from which the Origin was initialised changed, the change would no longer be reflected in Origin.